### PR TITLE
[CALCITE-4912] Confusing javadoc of RexSimplify.simplify

### PR DESCRIPTION
**Pull request by NobiGo**

### CODE REVIEW

File: core/src/main/java/org/apache/calcite/rex/RexSimplify.java [hunk 1]
```diff
@@ -204,8 +204,8 @@ public RexNode simplifyPreservingType(RexNode e, RexUnknownAs unknownAs,
    *
    * <p>In particular:</p>
    * <ul>
-   * <li>{@code simplify(x = 1 AND y = 2 AND NOT x = 1)}
```
Comment (wojustme) [inline core/src/main/java/org/apache/calcite/rex/RexSimplify.java:207]: Origin doc maybe means reduce condition of x.
Why you change here?

Comment (NobiGo) [reply core/src/main/java/org/apache/calcite/rex/RexSimplify.java:207]: It is a wrong simplify.

Comment (wojustme) [reply core/src/main/java/org/apache/calcite/rex/RexSimplify.java:207]: @NobiGo Yes, you are right.
My description may not be very accurate.
I mean that this expression should return FALSE
Should we just change return's doc?

Comment (NobiGo) [reply core/src/main/java/org/apache/calcite/rex/RexSimplify.java:207]: No. What the expression should be returned is decided by how we treat the UNKNOW. For example:
```
x=1 AND NOT x=1

If UNKNOW as FALSE then return FALSE;
If UNKNOW as TRUE  then return FALSE;
If UNKNOW as UNKNOW then return UNKNOW.(Then (x = 1 AND y = 2 AND NOT x = 1) result will not be a definite value)
```

Comment (wojustme) [reply core/src/main/java/org/apache/calcite/rex/RexSimplify.java:207]: Yes, It's my negligence
LTGM~

**Pull request merged**